mlir/test/Conversion/GPUToROCDL/rawbuf_store.mlir

```diff
@@ -0,0 +1,175 @@
+// RUN: mlir-opt %s -convert-gpu-to-rocdl='index-bitwidth=32' | FileCheck %s
+
+gpu.module @mubuf_store {
+  // f32 tests.
+
+  // CHECK-LABEL: func @buffer_store_f32_to_rank_1
+  func @buffer_store_f32_to_rank_1(%value : f32, %dst : memref<128xf32>, %shift : i32, %offset0 : i32) {
+    // CHECK: rocdl.buffer.store %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}} : f32
+    gpu.buffer_store(%value, %dst, %shift, %offset0) : f32, memref<128xf32>, i32
```

Comment:
turns out we are NOT really testing `gpu.raw_buffer_store` and `rocdl.raw.buffer.store` . We are considered lucky that the whole thing works.